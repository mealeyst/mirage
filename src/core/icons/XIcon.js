import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const BaseXIcon = ({className}) => {
  return (
    <svg className={className} xmlns='http://www.w3.org/2000/svg' viewBox='0 0 15 15'>
      <line x1='0' y1='0' x2='15' y2='15' />
      <line x1='0' y1='15' x2='15' y2='0' />
    </svg>
  )
}

const XIcon = styled(BaseXIcon)`
  width: ${props => props.width};
  line {
    stroke-width: 1.5;
    fill: none;
    stroke: ${props => props.stroke || props.theme.colors.rocketBlue};
  }
  cursor: pointer;
`

XIcon.propTypes = {
  theme: PropTypes.shape({
    colors: PropTypes.shape({
      rocketBlue: PropTypes.string
    })
  }),
  width: PropTypes.string
}

XIcon.defaultProps = {
  width: '10px'
}

/** @component */
export default XIcon
export { BaseXIcon }
