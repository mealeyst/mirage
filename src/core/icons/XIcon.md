```js
<XIcon />
```

## Resized Example:
```js
<XIcon width='15px' />
```
