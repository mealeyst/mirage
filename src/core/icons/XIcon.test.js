import React from 'react'
import { css } from 'styled-components'
import 'jest-styled-components'

import { XIcon } from 'SRC'
import { BaseXIcon } from './XIcon'

const { shallowWithTheme } = global

describe('(Styled Component) XIcon', () => {
  const createXIcon = (props) => {
    return shallowWithTheme(<XIcon {...props} />)
  }

  test('matching the snapshot', () => {
    expect(createXIcon())
    .toMatchSnapshot()
  })

  test('setting the width', () => {
    const width = '6rem'
    expect(createXIcon({width: width}))
    .toHaveStyleRule({
      width: width
    })
  })
})

describe('(Component) BaseXIcon', () => {
  const createBaseXIcon = (props) => {
    return shallowWithTheme(<BaseXIcon {...props} />)
  }

  test('matching the snapshot', () => {
    expect(createBaseXIcon())
    .toMatchSnapshot()
  })

  test('setting the className', () => {
    const className = 'example-class'
    expect(
      createBaseXIcon({className: className}).prop('className')
    ).toEqual(className)
  })
})
