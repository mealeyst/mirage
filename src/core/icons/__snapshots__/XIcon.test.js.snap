// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) BaseXIcon matching the snapshot 1`] = `
<svg
  viewBox="0 0 15 15"
  xmlns="http://www.w3.org/2000/svg"
>
  <line
    x1="0"
    x2="15"
    y1="0"
    y2="15"
  />
  <line
    x1="0"
    x2="15"
    y1="15"
    y2="0"
  />
</svg>
`;

exports[`(Styled Component) XIcon matching the snapshot 1`] = `
.c0 {
  width: 10px;
  cursor: pointer;
}

.c0 line {
  stroke-width: 1.5;
  fill: none;
  stroke: #0073D1;
}

<BaseXIcon
  className="c0"
  width="10px"
/>
`;
